SurfaceTopography/Nonuniform/Converters.py

```diff
@@ -118,52 +58,65 @@ def __init__(self, topography, nb_points=None, padding=0, nb_interpolate=None, i
             Number of grid points to between closest points on surface.
             (Default: None)
+        pixel_size : float, optional
+            Grid spacing. (Default: None)
         """
         super().__init__(topography, info=info)
         self.nb_points = nb_points
         self.nb_interpolate = nb_interpolate
-        self._update_nb_points()
         self.padding = padding
+        self.in_pixel_size = pixel_size
+        self._update_nb_points_and_pixel_size()
 
         # This is populated with functions from the nonuniform topography, but
         # this is a uniform topography
         self._functions = UniformLineScan._functions
 
-    def _update_nb_points(self):
-        """Automatically compute nb_points if it is None"""
-        if self.nb_points is None:
-            if self.nb_interpolate is None:
-                raise ValueError('You need to specify either `nb_points` or `nb_interpolate`.')
-            s, = self.parent_topography.physical_sizes
-            x = self.parent_topography.positions()
-            min_dist = np.min(np.diff(x))
-            if min_dist <= 0:
-                raise RuntimeError('This is a reentrant nonuniform line scan. Reentrant line scans cannot be converted '
-                                   'to uniform line scans.')
+    def _update_nb_points_and_pixel_size(self):
+        """Automatically compute `nb_points` and `pixel_size` if it is None"""
+        s, = self.parent_topography.physical_sizes
+        x = self.parent_topography.positions()
+        min_dist = np.min(np.diff(x))
+        if min_dist <= 0:
+            raise RuntimeError('This is a reentrant nonuniform line scan. Reentrant line scans cannot be converted '
+                               'to uniform line scans.')
+
+        if self.in_pixel_size is not None:
+            if self.nb_points is not None or self.nb_interpolate is not None:
+                raise ValueError('You need to specify either `nb_points`, `nb_interpolate` or `pixel_size`.')
+            self._nb_points = int(s / self.in_pixel_size) + 1
+            self._pixel_size = self.in_pixel_size
+        elif self.nb_interpolate is not None:
+            if self.nb_points is not None or self.in_pixel_size is not None:
+                raise ValueError('You need to specify either `nb_points`, `nb_interpolate` or `pixel_size`.')
             self._nb_points = self.nb_interpolate * int(s / min_dist)
-        else:
-            if self.nb_interpolate is not None:
-                raise ValueError('You cannot specify both, `nb_points` and `nb_interpolate`.')
+            self._pixel_size = s / self._nb_points
+        elif self.nb_points is not None:
+            if self.nb_interpolate is not None or self.in_pixel_size is not None:
+                raise ValueError('You need to specify either `nb_points`, `nb_interpolate` or `pixel_size`.')
             self._nb_points = self.nb_points
+            self._pixel_size = s / self._nb_points
```

Comment:
Shouldn't you divide by (nb_points - 1 ) to be consistent with above ? 

How should I interprete the pixels ? the space between the points or as centered around the points ?

Reply:
That is correct, there is a -1 that is missing, fixed it. There was also one missing below in the definition of `physical_sizes`.

The size of the domain actually changes when you turn a uniform line scan into a nonuniform one. This is because the interpretation is slightly different, per pixel vs. node points. (This interpretation is of course open for discussion.)

Reply:
ok. I can make a little svg to fix this in the doc

Reply:
That would probably be useful, but is not a priority